src/scripts/docker_migration_apt.sh

```diff
@@ -0,0 +1,168 @@
+curl -fL "${DOWNLOAD_GPG_URL}" |
+  gpg --yes --dearmor -o /etc/apt/keyrings/docker.gpg 2>&1 | tee -a ${LOG_FILE}
+if [ $? -ne 0 ]; then
+  log "Failed to download and install docker gpg key."
+  exit 1
+fi
+log "Change permissions for docker gpg key"
+chmod a+r /etc/apt/keyrings/docker.gpg 2>&1 | tee -a ${LOG_FILE}
+if [ $? -ne 0 ]; then
+  log "Failed to change permissions for docker gpg key."
+  exit 1
+fi
+# 3. Use the following command to set up the repository
+log "Add docker repository"
+echo \
+  "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/docker.gpg] ${DOWNLOAD_REPO_URL} \
+  $(source /etc/os-release && echo "$VERSION_CODENAME") stable" 2>&1 |
+  tee /etc/apt/sources.list.d/docker.list >/dev/null
+if [ $? -ne 0 ]; then
+  log "Failed to add docker repository."
+  exit 1
+fi
+
+# 1. Update the apt package index:
+log "Update the apt packages again"
+apt-get update 2>&1 | tee -a ${LOG_FILE}
+if [ $? -ne 0 ]; then
+  log "Failed to update"
+  exit 1
+fi
+# IMPORTANT: This step MUST be skipped so unattended-upgrades will upgrade docker later on
+# 2. Install Docker Engine, containerd, and Docker Compose.
+#log "Install docker-ce docker-ce-cli containerd.io docker-buildx-plugin docker-compose-plugin"
+#apt-get install docker-ce docker-ce-cli containerd.io docker-buildx-plugin docker-compose-plugin -y 2>&1 | tee -a ${LOG_FILE}
+#if [ $? -ne 0 ]; then
+#  log "Failed to install docker-ce docker-ce-cli containerd.io docker-buildx-plugin docker-compose-plugin."
+#  exit 1
+#fi
+# 3. Verify that the Docker Engine installation is successful by running the hello-world image.
+#docker run --rm hello-world && docker rmi hello-world
+
+# Add docker to unattended-upgrades
+log "Add docker to unattended-upgrades"
+# Check that the UNATTENDED_upgrades_file exists if so, check that the file does not already contain the DOCKER_DOWNLOAD_ORIGINS, if not then modify it to include in the section Unattended-Upgrade::Allowed-Origins the docker download origins
+if ! grep -q "${DOCKER_DOWNLOAD_ORIGINS}" "${UNATTENDED_UPGRADES_FILE}"; then
+  log "Add docker download origins to unattended-upgrades"
+  sed -i "/Unattended-Upgrade::Allowed-Origins {/a \"${DOCKER_DOWNLOAD_ORIGINS}\";" "${UNATTENDED_UPGRADES_FILE}" 2>&1 | tee -a ${LOG_FILE}
+fi
+
+# Add docker-compose alias of docker compose to the dappnode profile if is not already there and if docker compose is installed
+if [ -x "$(command -v docker-compose)" ]; then
+  if ! grep -q "alias docker-compose='docker compose'" /usr/src/dappnode/DNCORE/.dappnode_profile; then
+    log "Adding docker-compose alias to the dappnode profile"
+    echo "alias docker-compose='docker compose'" >>/usr/src/dappnode/DNCORE/.dappnode_profile
+    source /usr/src/dappnode/DNCORE/.dappnode_profile | tee -a ${LOG_FILE}
+    if [ $? -ne 0 ]; then
+      log "Failed to source dappnode profile."
+      exit 1
+    fi
+    # TODO: consider removing old docker-compose binary
```

Comment:
You are right here. Do we know which of them will be used if both exist ( `docker-compose` alias and `docker-compose` old binary)?